web/src/message_view_header.js

```diff
@@ -162,6 +135,8 @@ export function exit_search() {
         // for "searching narrows", we redirect
         window.location.href = filter.generate_redirect_url();
     }
+    // TODO(pre-merge): why does this not move the cursor out of the search box
```

Comment:
That might be a delay issue, especially after setting `window.location.href` above. You might experiment with a `setTimeout()` around the focus trigger.